src/push_context.rs

```diff
@@ -393,3 +385,125 @@ impl PushContext {
         Ok(ctx)
     }
 }
+
+fn get_project_owner_and_name(
+    repository: &String,
+    exec_env: ExecutionEnvironment,
+) -> Result<(String, String)> {
+    let mut repository_split = repository.split('/');
```

Comment:
What would you think about using `split_once('/')` in here instead, then doing `.replace('/', '-')` on the second value? That'll cover any number of `/`s cleanly.

Reply:
I'm going back and forth on this. Only GitLab supports this subgroup notion, and we know it's only 3 segments (group/subgroup/repo, ish), so we should be able to error on unexpected inputs and not unexpectedly allow malformed input to be made "safe"

Reply:
Yeah, I'm seeing now that subgroups can go pretty deep. I think I have a reasonable solution.

Reply:
It looks like this is still

Reply:
...using an unbound split, and erroring if there are more than 3 parts. I think it'd be cleaner to split just once, and operate on the right hand side all at once.

![image](https://github.com/DeterminateSystems/flakehub-push/assets/76716/01111e84-d05f-47ea-98d2-2c6517de83de)

Reply:
It only errors on more than three parts if subgroup renaming is disabled. Otherwise it supports an indefinite number of segments.

Reply:
You can see in the test cases that e.g. `a/b/c/d/e/f/g/h/i/j/k/l` resolves as expected

Reply:
Yeah, just looking at this my $0.02 is the implementation is quite a bit more complicated than it needs to be. Splitting once we can tell if they did the initial customer/project name correctly. From there, if there are `/`s do the string replacement or error if it is disabled.

Reply:
We can't really do string replacement here, though, because we're working with segments of a `Split`. I'm not sure I can make it much simpler than this.

Reply:
Ah okay, I see what you mean about the `upload_name` logic above. I'll fix it.